### PR TITLE
Update and rename main.sh

### PR DESCRIPTION
**Pull request by mawad89**

### CODE REVIEW

**Review by drtamermansour — COMMENTED** [review]
Please do not change the file name. If you want to keep multiple versions of files, use different branches in Github.

File: main_27_7_2018.sh [hunk 1]
```diff
@@ -34,26 +38,50 @@ sample_accessions=("SRR1258218" "SRR1153470")
...
 
 ## download GIAB data using fastq-dump
 cd $work_dir/data
 module load GNU/4.4.5
 module load SRAToolkit/2.8.2
-for sample_accession in ${sample_accession[@]};do
-  fastq-dump --defline-seq '@$sn[_$rn]/$ri' --split-files $sample_accession
-done
+wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR115/SRR1153470/SRR1153470.sra
+wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR125/SRR1258218/SRR1258218.sra
```
Comment (drtamermansour) [inline main_27_7_2018.sh]: I am not sure why did you replace the direct dump by wget/dump but it is OK. You still can use the for-loop on sample accessions (but you need to change ${sample_accession[@]} to ${sample_accessions[@]} to fix a bug). Inside the for-loop, you cab wget, dump, then rm the sra

File: main_27_7_2018.sh [hunk 2]
```diff
@@ -34,26 +38,50 @@ sample_accessions=("SRR1258218" "SRR1153470")
...
 wget $ref_vcf_url -O $work_dir/ref_vcf/$sample.vcf.gz
+gunzip $ref_vcf/$sample.vcf.gz#gunzip
```
Comment (drtamermansour) [inline main_27_7_2018.sh]: remove the comment (In the general, there should be a space before the hash character)

File: main_27_7_2018.sh [hunk 3]
```diff
@@ -34,26 +38,50 @@ sample_accessions=("SRR1258218" "SRR1153470")
 ref_vcf_url="ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/NA12878_HG001/latest/GRCh38/HG001_GRCh38_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-X_v.3.3.2_highconf_PGandRTGphasetransfer.vcf.gz"
 ref_genome="ens_build38_release92"
 ref_genome_url='ftp://ftp.ensembl.org/pub/release-92/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.*.fa.gz' ## primaly assembly has unplaced contigs while the vcf has only chromosomes
+ref_annot='ftp://ftp.ensembl.org/pub/release-92/gtf/homo_sapiens/Homo_sapiens.GRCh38.92.gtf.gz'
```
Comment (drtamermansour) [inline main_27_7_2018.sh]: You need to have two variables: ref_annot & ref_annot_url. The url will be used for download only but the 1st variable should refer to the file itself  e.g.
ref_annot="Homo_sapiens.GRCh38.92.gtf"

File: main_27_7_2018.sh [hunk 4]
```diff
@@ -34,26 +38,50 @@ sample_accessions=("SRR1258218" "SRR1153470")
...
+cd $work_dir
+mkdir -p STAR
+STAR --genomeDir $outDir/ --readFilesIn $data/${sample_accession}_subset_1.fastq,$data/${sample_accession}_subset_2.fastq --outFileNamePrefix STAR/${sample_accession}_supset --sjdbGTFfile $anno_dir/* --runThreadN 7
+
+#MarkDuplicates (Picard)
+cd $work_dir
+mkdir -p picard
+#there is no output from this file
+java -jar $picard MarkDuplicates I=STAR/${sample_accession}_supsetAligned.out.bam O=picard/marked_duplicates_${sample_accession}_subset.bam M=picard/marked_dup_metrics_${sample_accession}_subset.txt
+
```
Comment (drtamermansour) [inline main_27_7_2018.sh]: As we discussed before, this block of code should be after line 63 and use only the variables defined in lines 3-10 and 59-61

File: main_27_7_2018.sh [hunk 5]
```diff
@@ -34,26 +38,50 @@ sample_accessions=("SRR1258218" "SRR1153470")
...
 ## Varinat Calling 
 ## Change these variable to define the input files
 sample="NA12878"                     # the reference VCF will be  $ref_vcf/$sample.vcf.gz
```
Comment (drtamermansour) [inline main_27_7_2018.sh]: Again, you need to copy the ref_annot variable used for this analysis here

File: main_27_7_2018.sh (hunk 4 above)
Comment (mawad89) [reply main_27_7_2018.sh]: where is the problem

**Comment by drtamermansour** [comment]
- TM: There is only one bug that needs to be fixed at line 42 in the original file.
- TM: then you should write your code after line 63
- TM: you do not need to write code for all samples. Only the sample defined in lines 59-61 with its genome
- MA: In addition to some variables  we already declare it but we rewrite it again
TM: No. no need to declare any new variables. Just use the variables defined in lines 3-9 and lines 59-61
- TM: forget about the data download section
MA: Ok I also downloaded the annotation
TM: Make sense. you can add two lines after line 36 to define the annotation variable and its download url. Then download and unzip (if needed after line 50).
Also, copy your new line which defines the annotation variable after line 61. This should be exactly matching to what we did with ref_genome


previous notes have not been addressed completely yet

File: main.sh [hunk 6]
```diff
@@ -34,26 +38,51 @@ sample_accessions=("SRR1258218" "SRR1153470")
...
+wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR115/SRR1153470/SRR1153470.sra
+wget ftp://ftp-trace.ncbi.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR125/SRR1258218/SRR1258218.sra
 wget $ref_vcf_url -O $work_dir/ref_vcf/$sample.vcf.gz
+gunzip $ref_vcf/$sample.vcf.gz
 wget $ref_vcf_url.tbi -O $work_dir/ref_vcf/$sample.vcf.gz.tbi
 wget $ref_genome_url -P $work_dir/genomeDir
 gunzip -c $work_dir/genomeDir/Homo_sapiens.GRCh38.dna.chromosome.*.fa.gz > $work_dir/genomeDir/$ref_genome.fa
 rm $work_dir/genomeDir/Homo_sapiens.GRCh38.dna.chromosome.*.fa.gz 
 sed -i 's/^>\(.*\) dna.*/>chr\1/' $work_dir/genomeDir/$ref_genome.fa
+wget $ref_annot_url -O $anno_dir
```
Comment (drtamermansour) [inline main.sh:57]: you can use the -P option instead to avoid moving the file after download

**Pull request closed without merging**